### PR TITLE
Variety of fixes for 2023 beta

### PR DESCRIPTION
**Pull request by virtuald**

### CODE REVIEW

File: robotpy_build/platforms.py [hunk 1]
```diff
@@ -62,7 +62,11 @@ def get_platform(name: typing.Optional[str] = None) -> WPILibMavenPlatform:
         pyplatform = _get_platform()
 
         # Check for 64 bit x86 macOS (version agnostic)
-        if re.fullmatch(r"macosx-.*-x86_64", pyplatform):
+        # - See https://github.com/pypa/setuptools/issues/2520 for universal2
```
Comment (virtuald) [inline robotpy_build/platforms.py:65]: This is almost certainly going to lead to problems, but not sure how to deal with it for now.

File: robotpy_build/hooks.py [hunk 2]
```diff
@@ -842,3 +878,35 @@ def class_hook(self, cls, data):
         cls["x_varname"] = "cls_" + cls_name
         cls["x_name"] = self._set_name(cls_name, class_data)
         cls["x_doc_quoted"] = self._process_doc(cls, class_data)
+        cls["x_inline_code"] = class_data.inline_code if class_data.inline_code else ""
```
Comment (auscompgeek) [inline robotpy_build/hooks.py:881]: could do
```suggestion
        cls["x_inline_code"] = class_data.inline_code or ""
```

File: robotpy_build/platforms.py (hunk 1 above)
Comment (auscompgeek) [reply robotpy_build/platforms.py:65]: We should probably look into supporting native ARM at some point, iirc WPILib is building for ARM for next season

File: robotpy_build/tool.py [hunk 3]
```diff
@@ -127,39 +127,67 @@ def add_subparser(cls, parent_parser, subparsers):
...
+                    if lastdir is None:
+                        if thisdir:
+                            print("#", thisdir)
+                    elif lastdir != thisdir:
+                        print()
+                        if thisdir:
+                            print("#", thisdir)
+                    lastdir = thisdir
+
+                    base = splitext(f.name)[0]
```
Comment (auscompgeek) [inline robotpy_build/tool.py:189]: I think this is equivalent?
```suggestion
                    base = f.stem
```
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem

**Pull request merged**